Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -193,17 +193,18 @@ public async Task<ContestDetailsServiceModel> GetContestDetails(int id)
                 requiredPasswordIsValid = true;
             }
 
+            bool shouldNotRequirePasswordAndIsNotOnline = !registerModel.RequirePassword && !contest.IsOnlineExam;
+            bool shouldNotRequirePasswordAndIsNotOfficial = !registerModel.RequirePassword && !isOfficial;
+            bool requiredPasswordIsValidAndIsNotOnline = requiredPasswordIsValid && !contest.IsOnlineExam;
```

Comment:
If the contest has password, why IsOnline should be a factor at all?

Reply:
In the case of an online contest, shouldConfirmParticipation should be checked too.